src/Functions/minSampleSize.cpp

```diff
@@ -0,0 +1,247 @@
+            {
+                if (!isNativeNumber(arguments[i]))
+                {
+                    throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT, "The {}th Argument of function {} must be a native number.", i + 1, name);
+                }
+            }
+
+            for (size_t i = 2; i < arguments.size(); ++i)
+            {
+                if (!isFloat(arguments[i]))
+                {
+                    throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT, "The {}th Argument of function {} must be a float.", i + 1, name);
+                }
+            }
+        }
+
+        static ColumnPtr execute(const ColumnsWithTypeAndName & arguments, size_t input_rows_count)
+        {
+            const IColumn & col_baseline = *arguments[0].column;
+            const IColumn & col_sigma = *arguments[1].column;
+            const IColumn & col_mde = *arguments[2].column;
+            const IColumn & col_power = *arguments[3].column;
+            const IColumn & col_alpha = *arguments[4].column;
+
+            auto res_min_sample_size = ColumnFloat64::create();
+            auto & data_min_sample_size = res_min_sample_size->getData();
+            data_min_sample_size.reserve(input_rows_count);
+
+            auto res_detect_lower = ColumnFloat64::create();
+            auto & data_detect_lower = res_detect_lower->getData();
+            data_detect_lower.reserve(input_rows_count);
+
+            auto res_detect_upper = ColumnFloat64::create();
+            auto & data_detect_upper = res_detect_upper->getData();
+            data_detect_upper.reserve(input_rows_count);
+
+            for (size_t row_num = 0; row_num < input_rows_count; ++row_num)
+            {
+                /// Mean of control-metric
+                Float64 baseline = col_baseline.getFloat64(row_num);
+                /// Standard deviation of conrol-metric
+                Float64 sigma = col_sigma.getFloat64(row_num);
+                /// Minimal Detectable Effect
+                Float64 mde = col_mde.getFloat64(row_num);
+                /// Sufficient statistical power to detect a treatment effect
+                Float64 power = col_power.getFloat64(row_num);
+                /// Significance level
+                Float64 alpha = col_alpha.getFloat64(row_num);
+
+                if (!std::isfinite(baseline) || !std::isfinite(sigma) || !isBetweenZeroAndOne(mde) || !isBetweenZeroAndOne(power) || !isBetweenZeroAndOne(alpha))
+                {
+                    data_min_sample_size.emplace_back(std::numeric_limits<Float64>::quiet_NaN());
+                    data_detect_lower.emplace_back(std::numeric_limits<Float64>::quiet_NaN());
+                    data_detect_upper.emplace_back(std::numeric_limits<Float64>::quiet_NaN());
+                    continue;
+                }
+
+                Float64 delta = baseline * mde;
+
+                using namespace boost::math;
+                normal_distribution<> nd(0.0, 1.0);
```

Comment:
It seems that distribution constructor should be moved out of hot loop.

Reply:
I moved distribution constructor out of the hot loop.